src/_pytest/compat.py

```diff
@@ -342,39 +352,26 @@ def safe_isclass(obj):
 )
 
 
-def _setup_collect_fakemodule():
+def _setup_collect_fakemodule() -> None:
     from types import ModuleType
     import pytest
 
-    pytest.collect = ModuleType("pytest.collect")
-    pytest.collect.__all__ = []  # used for setns
+    # Types ignored because the module is created dynamically.
+    pytest.collect = ModuleType("pytest.collect")  # type: ignore
+    pytest.collect.__all__ = []  # type: ignore  # used for setns
     for attr_name in COLLECT_FAKEMODULE_ATTRIBUTES:
-        setattr(pytest.collect, attr_name, getattr(pytest, attr_name))
+        setattr(pytest.collect, attr_name, getattr(pytest, attr_name))  # type: ignore
 
 
 class CaptureIO(io.TextIOWrapper):
-    def __init__(self):
+    def __init__(self) -> None:
         super().__init__(io.BytesIO(), encoding="UTF-8", newline="", write_through=True)
 
-    def getvalue(self):
+    def getvalue(self) -> str:
+        assert isinstance(self.buffer, io.BytesIO)
         return self.buffer.getvalue().decode("UTF-8")
 
 
-class FuncargnamesCompatAttr:
-    """ helper class so that Metafunc, Function and FixtureRequest
-    don't need to each define the "funcargnames" compatibility attribute.
-    """
-
-    @property
-    def funcargnames(self):
-        """ alias attribute for ``fixturenames`` for pre-2.3 compatibility"""
-        import warnings
-        from _pytest.deprecated import FUNCARGNAMES
-
-        warnings.warn(FUNCARGNAMES, stacklevel=2)
-        return self.fixturenames
-
-
```

Comment:
I'm ok with inlining, but wondered how it made typing harder (from the commit message)?

Reply:
It's a mixin class, it uses `fixturenames` which is not defined in the class so mypy gets confused. There are ways to type this but it's not worth the bother in this case IMO.

Reply:
Agreed.